src/AppBundle/Entity/Setting.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+namespace AppBundle\Entity;
+
+use Doctrine\ORM\Mapping as ORM;
+use JMS\Serializer\Annotation as JMS;
+
+/**
+ * @ORM\Entity
+ * @ORM\Table(name="setting")
+ */
+class Setting
+{
+    /**
+     * @var int
+     * @JMS\Type("string")
+     * @JMS\Groups({"setting"})
+     *
+     * @ORM\Column(name="id", type="string", length=64, nullable=false)
+     * @ORM\Id
+     */
+    private $id;
```

Comment:
I would keep the id as an integer and have a settings_type field which could be a string

Reply:
I thought the same, but then I'd need to make it more complicated with the getBy() and find(). ID autoincrement not strictly necessary for a fixed list of settings. What about we review this next time we need to add another setting ? autoincrement Id are not mandatory in doctrine. Ids are, and whether they are strings or int, make no difference in the end

Reply:
Im not sure why would it be more complicated? Personally, I find it more complicated to see an id being represented as a string when all other ids are integers. Hence I was confused when i read the tests and cross referenced with the entity doc blocks. Wasnt even sure how it was working at first.
Calls to getBy and find() should simply use :
findOneBy(array('settings_type' => $type);